src/platform/intel/cavs/lib/pm_runtime.c

```diff
@@ -62,6 +62,26 @@ static inline void cavs_pm_runtime_force_host_dma_l1_exit(void)
 	spin_unlock_irq(_prd->lock, flags);
 }
 
+static inline void cavs_pm_runtime_request_dsp_d0(uint32_t req)
+{
+	uint32_t flags;
+	struct cavs_pm_runtime_data *pprd =
+			(struct cavs_pm_runtime_data *)_prd->platform_data;
+
+	spin_lock_irq(_prd->lock, flags);
```

Comment:
If it's never accessed by the slave cores we could just use critical section instead of the full locking:
```
#include <sof/drivers/interrupt.h>
...
irq_local_disable(flags);
...
irq_local_enable(flags);
```

Reply:
Replaced the spin lock.